feature-libs/asm/components/asm-main-ui/asm-main-ui.component.ts

```diff
@@ -160,32 +174,32 @@ export class AsmMainUiComponent implements OnInit {
   }
 
   showCustomList(): void {
-    this.modalRef = this.modalService?.open(CustomerListComponent, {
-      centered: true,
-      size: 'mf',
-      windowClass: 'fiori-like',
-      ariaLabelledBy: 'asm-customer-list-title',
-      ariaDescribedBy: 'asm-customer-list-desc',
-    });
-    this.modalRef?.result
-      .then(({ selectedUser, actionType }: CustomerListAction) => {
-        if (selectedUser) {
-          this.startCustomerEmulationSession(selectedUser);
-          if (actionType === CustomerListColumnActionType.ORDER_HISTORY) {
-            this.routingService.go({ cxRoute: 'orders' });
+    this.launchDialogService?.openDialogAndSubscribe(
+      LAUNCH_CALLER.ASM_CUSTOMER_LIST,
+      this.element
+    );
+
+    this.subscription.add(
+      this.launchDialogService?.dialogClose
```

Comment:
each time function `showCustomList` is called, a new subscription will be created. So, the code block `    this.subscription.add(...)` can be moved into `ngOnInit`, right?

Reply:
thanks, you are right we can move into ngOnInit.